File: spring-springmvc-hibernate/src/main/java/com/jiangKlijna/web/control/UserControl.java
package com.jiangKlijna.web.control;

import javax.annotation.Resource;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import com.jiangKlijna.web.bean.Result;
import com.jiangKlijna.web.service.UserService;

@RequestMapping("/user")
@Controller
public class UserControl extends BaseControl {

    @Resource(name = "userService")
    public UserService us;

    @RequestMapping("/regist")
    public void regist(String username, String password, String action) {
        Result re = testParameter(username, password, action) ? us.regist(username, password) : errorParameterResult;
        switch (action) {
            case "html":
                html(re.setData(username + password).toString());
                break;
            case "json":
                json(re.setData(username + password));
                break;
            case "xml":
                xml(re.setData(username + password));
                break;
            default:
            case "text":
                text(re.setData(username + password).toString());
                break;
        }
    }

    @RequestMapping("/remove")
    public void remove(int id) {
        Result re = us.remove(id);
        xml(re.setData(id));
    }

    @RequestMapping("/find")
    public void find(int id) {
        xml(us.find(id));
    }
}

File: springmvc/spring-springmvc-springdatajpa-kotlin/src/main/java/com/jiangKlijna/web/control/UserControl.kt
package com.jiangKlijna.web.control

import javax.annotation.Resource

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping

import com.jiangKlijna.web.bean.Result
import com.jiangKlijna.web.service.UserService
import org.springframework.web.bind.annotation.ResponseBody

@RequestMapping("/user")
@Controller
class UserControl : BaseControl() {

    @Resource(name = "userService")
    var us: UserService? = null

    @ResponseBody
    @RequestMapping("/regist.json")
    fun regist_json(username: String, password: String): Result {
        return if (BaseControl.testParameter(username, password)) us!!.regist(username, password) else BaseControl.errorParameterResult
    }

    @ResponseBody
    @RequestMapping("/regist.xml")
    fun regist_xml(username: String, password: String): Result {
        return if (BaseControl.testParameter(username, password)) us!!.regist(username, password) else BaseControl.errorParameterResult
    }

    @ResponseBody
    @RequestMapping("/remove.json")
    fun remove(id: Int): Result = us!!.remove(id)

    @ResponseBody
    @RequestMapping("/find.xml")
    fun find(id: Int): Result = us!!.find(id)
}
